packages/nimble-components/src/table/index.ts

```diff
@@ -268,19 +271,55 @@ export class Table<
         );
     }
 
+    private updateTanStack(): void {
+        const updatedOptions: Partial<TanStackTableOptionsResolved<TData>> = {
+            state: {}
+        };
+
+        if (this.updateTracker.updateColumnSort) {
```

Comment:
The alternative to this series of `if` statements is to put that branching logic into the individual methods, like the following:
```
private getTanStackSortState() {
    if (this.updateTracker.updateColumnSort) {
        ...
        return new state;
    }
    return this.options.state.sorting;
}
```

This may also be good from a semantic point of view, in that if this method was ever called from some other context it wouldn't do unnecessary work.

Reply:
Another option is that I could leave the code as-is, but rename all the `getTanStackXYZ()` functions to something like `calculateTanStackXYZ()` to make it more obvious from the function calls that it is not a simple getter. Then, if we found that we have a need to have logic like you suggest, we could add back the `getTanStackXYZ()` functions that have conditional logic between calling `calculateTanStackXYZ()` and returning the current option.

Thoughts?

Reply:
I like that approach.